jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/WebAppProvider.java

```diff
@@ -265,8 +266,18 @@ public ContextHandler createContextHandler(final App app) throws Exception
         if (!resource.exists())
             throw new IllegalStateException("App resource does not exist " + resource);
 
-        String context = file.getName();
+        final String contextName = file.getName();
 
+        // Resource aliases (after getting name) to ensure baseResource is not an alias
+        if (resource.isAlias())
+        {
+            resource = Resource.newResource(resource.getAlias());
+            file = resource.getFile().toPath().toRealPath().toFile();
```

Comment:
Do we really need the `toRealPath()` here, shouldn't the `resource.getAlias()` already give you this?

Reply:
Maybe not, but I wanted to protect against changes to the implemention of Resource and also to deal with things like symlinks to symlinks.   Better to be safe than sorry.

Reply:
Maybe then you should do something like:
```java
file = new File(resource.getAlias()).toPath().toRealPath().toFile();
resource = Resource.newResource(file);
```

so that `file` and `resource.getFile()` are the same.

Reply:
very good point!